lib/wallet/http.js

```diff
@@ -1246,6 +1247,83 @@ class HTTP extends Server {
       return res.json(200, mtx.getJSON(this.network));
     });
 
+    // Batch Bid
+    this.post('/wallet/:id/batch/bid', async (req, res) => {
+      const MAX_BIDS_ARRAY_LENGTH = 200;
+      const valid = Validator.fromRequest(req);
+      const passphrase = valid.str('passphrase');
+
+      if (!valid.has('bids'))
+         throw new Error('Mandatory Bids parameter is missing!');
+
+      let bids = valid.array('bids');
+      assert(bids.length > 0 && bids.length <= MAX_BIDS_ARRAY_LENGTH,
+        `Bids are required to be non empty and shoud not exceed 
+        ${MAX_BIDS_ARRAY_LENGTH}`);
+
+      bids.forEach((element) => {
+        const validator = new Validator(element);
+        if (!validator.has('name') ||
+            !validator.has('bid')  ||
+            !validator.has('lockup') ||
+            !validator.has('idempotencyKey')) {
+          throw new Error('Bids contains a Bid with missing mandatory fields!');
+        } else { // check types
+          validator.str('name');
+          validator.u64('bid');
+          validator.u64('lockup');
+          validator.str('idempotencyKey');
+        }
+      });
+
+     // add value property (same as bid) to each bid
+     // value property is same with bid property and used internally
+     bids = bids.map((element) => {
+        element['value'] = element.bid;
+        return element;
+      });
+
+      const options = TransactionOptions.fromValidator(valid);
+
+      // if already exists in the cache do not process it
+      let results = [];
+      let errors  = [];
+
+      const filteredBids = bids.filter((bid) => {
+          const bidResultFromCache = req.wallet
+          .sendBatchBidResults.get(bid.idempotencyKey);
+
+          if (bidResultFromCache != null) {
+              bidResultFromCache['fromCache'] = true;
+              results.push(bidResultFromCache);
+          }
+
+          return (bidResultFromCache == null);
+      });
+
+      if (filteredBids.length > 0) {
+        const {mtx, errorMessages} = await req.wallet
+        .createBatchBid(filteredBids, options);
+
+        errors = errorMessages;
+
+        // always broadcast and sign
+        const tx = await req.wallet.sendMTX(mtx, passphrase);
+        const txJSON = tx.getJSON(this.network);
+
+        const processedBids = util.postProcessBatchBids(txJSON, mtx);
```

Comment:
Could you explain why passing in both is necessary I wasn't able to get it from the util

Reply:
it gets the idempotency_key from the mtx and rest from txJSON, it transforms the bid outputs as below:
hsd-proxy uses this idempotency_key to populate it's own cache as well.

```
processedBids.push({
        idempotency_key: mtxOutput.idempotencyKey,
        tx_hash: txJSON.hash,
        output_index: i,
        output: txOutput
      });
```

Reply:
Ok if only the idempotency key is necessary, then can we just pass that into postProcessBatchBids instead of the whole mtx? I think it introduces confusion when there are two tx objects passed into that function